src/Nethermind/Nethermind.BeaconNode.Peering/MothraPeeringWorker.cs

```diff
@@ -212,6 +236,53 @@ await JsonSerializer.SerializeAsync(fileStream, signedBeaconBlock, _jsonSerializ
             }
         }
 
+        private async void HandlePeerDiscoveredAsync(string peerId)
+        {
+            try
+            {
+                if (_peerManager.AddPeer(peerId))
+                {
+                    // Delay to ensure connection is established
+                    await Task.Delay(TimeSpan.FromMilliseconds(500)).ConfigureAwait(false);
+
+                    await _synchronizationManager.OnPeerDialOutConnected(peerId).ConfigureAwait(false);
```

Comment:
this might be useful to setup a call where I show how the sync is done in Eth1 Nethermind - sync is one of the most complex part of it and maybe some of the experiences that we had would be useful

the hardcoded delay here looks strange - we should know exactly when the connection is established

Reply:
It was probably me trying to understand how Mothra works, and I thought it was a timing issue, i.e. Mothra triggering an event when the connection was opened before setting it up properly. Now that it is working, I will try removing the delay.

Reply:
REmoved